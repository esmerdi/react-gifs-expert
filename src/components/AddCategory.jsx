import { useState } from 'react'
import PropTypes from 'prop-types';


export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');
    
    const onSubmit = (event) => {
        event.preventDefault();
        if(inputValue.trim() <= 1) return;
        onNewCategory(inputValue.trim())
        setInputValue('')
    }

    return (
        <form onSubmit={e => onSubmit(e)} aria-label='form'>
            <input
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={e => setInputValue(e.target.value)}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}
