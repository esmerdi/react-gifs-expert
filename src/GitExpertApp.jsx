import { useState } from "react"
import { AddCategory, GifGrid } from "./components";

export const GitExpertApp = () => {

    const [categories, setCategories] = useState(['Dragon Ball']);

    const onAddCategory = (newCategory) => {
        if (categories.some(cat => cat.toLowerCase() === newCategory.toLowerCase())) return;
        setCategories([newCategory, ...categories])
    }

    return (
        <>
            <h1>GitExpertApp</h1>

            <AddCategory
                onNewCategory={(value) => onAddCategory(value)}
            />

            {
                categories.map(category =>(
                    /* estos () signigican que estamos retornando un 
                    nuevo objeto*/
                    <GifGrid key={category} category={category} />
                ))
            }

        </>
    )
}

export default GitExpertApp
