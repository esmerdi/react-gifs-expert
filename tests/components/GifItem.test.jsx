import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components"

describe('Pruebas en <GifItem.jsx/>', () => {

    const title = 'Saitama';
    const url = 'https://one-ponch.com/saitama.jpg';

    test('Debe hacer Match con el snapshot', () => {

        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();

    })

    test('Debe de mostrar la imagen con el URL y ALT indicado', () => {

        render(<GifItem title={title} url={url} />);
        //screen.debug();//para mostrar nuestro objeto de prueba.

        //expect(screen.getByRole('img').src).toBe(url);

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);

    });
    
    test('debe mostrar el titulo del componente', () => { 

        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();

     })

})