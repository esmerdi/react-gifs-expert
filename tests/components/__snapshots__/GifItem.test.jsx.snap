// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem.jsx/> Debe hacer Match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-ponch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
